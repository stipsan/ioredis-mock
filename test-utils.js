// Runs both a regular string suite and a buffer suite
export function runTwinSuite(commandName, cb) {
  const commands = [commandName, `${commandName}Buffer`]

  return commands.forEach(command =>
    command === commandName
      ? cb(command, (a, b) => a === b)
      : cb(command, (a, b) => a.equals(Buffer.from(b)))
  )
}
