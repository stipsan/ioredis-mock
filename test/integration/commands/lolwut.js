import Redis from 'ioredis'

// eslint-disable-next-line import/no-relative-parent-imports
import { runTwinSuite } from '../../../test-utils'

runTwinSuite('lolwut', command => {
  ;(process.env.IS_BROWSER && command.endsWith('Buffer')
    ? describe.skip
    : describe)(command, () => {
    const redis = new Redis()

    afterAll(() => {
      redis.disconnect()
    })

    test('should return Redis version by default', async () => {
      const result = await redis[command]()
      expect(result).toMatchSnapshot()
    })

    test('version 6', async () => {
      const result = await redis[command]('version', 6, 20, 10)
      expect(Buffer.isBuffer(result) ? result.toString() : result).toMatch(
        'hikikomori'
      )
    })

    test('version 5', async () => {
      const result = await redis[command]('version', 5, 20, 10)
      expect(Buffer.isBuffer(result) ? result.toString() : result).toMatch(
        'schotter'
      )
    })
  })
})
