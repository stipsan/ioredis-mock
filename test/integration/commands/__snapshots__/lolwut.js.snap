// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lolwut should return Redis version by default 1`] = `
"Redis ver. 7.2.1
"
`;

exports[`lolwutBuffer should return Redis version by default 1`] = `Buffer<52 65 64 69 73 20 76 65 72 2e 20 37 2e 32 2e 31 0a>`;
